src/js/view/controls/settings-menu.js

```diff
@@ -43,7 +46,8 @@ export function createSettingsMenu(controlbar, onVisibility, localization) {
                 // Activate the first submenu if clicking the default button
                 settingsMenu.activateFirstSubmenu(nonKeyboardInteraction);
             }
-            settingsMenu.open(isDefault, event);
+        
+            delayedOpen(isDefault, event);
```

Comment:
i understand the fix as is but i wonder if this is a case where event propagation is causing the multiple firings. this line in particular 
> clicking a button that opens a submenu triggers 2-3 tap events.
makes me wonder if it's just bubbling and an easier fix would be to `stopPropagation`? maybe you've already tried it

Reply:
As per our convo, stopPropogation won't help here because it's a single listener on a specific button causing the issue, it's not bubbling. PreventDefault isn't applicable without performance issues since we would have to set ui.passive to false, which would decrease performance of all event handling on settings menu.